lms/djangoapps/ccx/migrations/0004_seed_forum_roles_in_ccx_courses.py

```diff
@@ -0,0 +1,71 @@
+
+from django_comment_common.models import FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_MODERATOR, \
+    FORUM_ROLE_COMMUNITY_TA, FORUM_ROLE_STUDENT
+from django_comment_common.utils import STUDENT_ROLE_PERMISSIONS,  MODERATOR_ROLE_PERMISSIONS, \
+    ADMINISTRATOR_ROLE_PERMISSIONS
+
+
+def seed_forum_roles_for_existing_ccx(apps, schema_editor):
+    """
+    Seed forum roles and make CCX coach forum admin for respective CCX(s).
+
+    Arguments:
+        apps (Applications): Apps in edX platform.
+        schema_editor (SchemaEditor): For editing database schema i.e create, delete field (column)
+
+    """
+    CustomCourseForEdX = apps.get_model("ccx", "CustomCourseForEdX")
+    Role = apps.get_model("django_comment_common", "Role")
+    Permission = apps.get_model("django_comment_common", "Permission")
+    db_alias = schema_editor.connection.alias
+
+    for ccx in CustomCourseForEdX.objects.using(db_alias).all():
+        if not ccx.course_id or ccx.course_id.deprecated:
+            # prevent migration for deprecated course ids or invalid ids.
+            continue
+
+        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
+
+        # Create forum roles.
+        role_query_set = Role.objects.using(db_alias)
+        admin_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_ADMINISTRATOR, course_id=ccx_locator)
+        moderator_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_MODERATOR, course_id=ccx_locator)
+        community_ta_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_COMMUNITY_TA, course_id=ccx_locator)
+        student_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_STUDENT, course_id=ccx_locator)
+
+        # Add permissions for each role.
+        permission_query_set = Permission.objects.using(db_alias)
+        for name in ADMINISTRATOR_ROLE_PERMISSIONS:
+            admin_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for name in MODERATOR_ROLE_PERMISSIONS:
+            moderator_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for name in STUDENT_ROLE_PERMISSIONS:
+            student_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for permission in student_role.permissions.all():
+            moderator_role.permissions.add(permission)
+        for permission in moderator_role.permissions.all():
+            community_ta_role.permissions.add(permission)
+        for permission in moderator_role.permissions.all():
+            admin_role.permissions.add(permission)
+
+        # Make CCX coach forum admin.
+        ccx.coach.roles.add(admin_role)
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('ccx', '0003_add_master_course_staff_in_ccx'),
+        ('django_comment_common', '0002_forumsconfig'),
+    ]
+
+    operations = [
+        migrations.RunPython(code=seed_forum_roles_for_existing_ccx, reverse_code=migrations.RunPython.noop)
```

Comment:
I've included `reverse_code=migrations.RunPython.noop` here to indicate that the migration is reversible, but nothing needs to be done to revert it.  Otherwise, the migration is considered irreversible, and trying to revert it results in an error.